src/widget.cpp

```diff
@@ -1754,7 +1754,8 @@ std::string format_widget_multiline( const std::vector<std::string> &keys, int m
                 }
             }
         }
-        if( row < h_max - 1 ) {
+        // Newline, if not the last row, and still keys left
+        if( row < h_max - 1 && nidx < nsize ) {
```

Comment:
@dseguin Tests are failing now due to the missing trailing newline. I saw you wrote the code and test. Is there a specific reason for the training newline?

Reply:
No reason for the extra newline, you can remove them from the tests. I think it wasn't an issue before because the next panel would be drawn over the empty row, so I never noticed it, ahah

Reply:
Thanks! Modified the tests accordingly.

However, widget strings now seem to exactly match length with the given width. So spaces are added to elongate, or ellipsis to shorten.

Seems consistent to me, but differs from the old behaviour.